core/src/blocktree.rs

```diff
@@ -193,6 +193,46 @@ impl Blocktree {
         false
     }
 
+    // Helper that deletes columns that have `(Slot,Index)` as their Index
+    fn delete_indexed_column<C>(&self, column: &LedgerColumn<C>, delete_slot: u64) -> Result<()>
+    where
+        C: Column<Index = (u64, u64)>,
+    {
+        let iter = column.iter(Some((delete_slot, 0)))?;
+        let mut max_index = 0;
+
+        for ((slot, index), _) in iter {
+            if slot != delete_slot {
+                break;
+            }
+            max_index = index;
+        }
+        for i in 0..=max_index {
+            column.delete((delete_slot, i))?;
+        }
+        Ok(())
+    }
+
+    fn delete_slot(&self, slot: u64) -> Result<()> {
+        match self.meta(slot) {
+            Ok(meta) => match meta {
+                None => Ok(()),
+                Some(_) => {
+                    self.meta_cf.delete(slot)?;
+                    self.delete_indexed_column(&self.data_cf, slot)?;
+                    self.delete_indexed_column(&self.erasure_meta_cf, slot)?;
```

Comment:
why would we bail on io errors in this prune?  better to bail on a consistency check?

Reply:
I will log all errors and attempt continue with the deletion.